pkg/cmd/auth/shared/login_flow.go

```diff
@@ -89,11 +92,28 @@ func Login(opts *LoginOptions) error {
 			if keyChoice < len(pubKeys) {
 				keyToUpload = pubKeys[keyChoice]
 			}
-		} else {
-			var err error
-			keyToUpload, err = opts.sshContext.generateSSHKey()
+		} else if opts.sshContext.HasKeygen() {
+			var sshChoice bool
+			err := prompt.SurveyAskOne(&survey.Confirm{
+				Message: "Generate a new SSH key to add to your GitHub account?",
+				Default: true,
+			}, &sshChoice)
+
 			if err != nil {
-				return err
+				return fmt.Errorf("could not prompt: %w", err)
+			}
+
+			if sshChoice {
+				passphrase, err := promptForSshKeyPassphrase()
+				if err != nil {
+					return err
```

Comment:
Since we have several places in this method where we just `return err` adding some context would be nice.

Reply:
The message in `promptForSshKeyPassphrase` already wrapped the error in a good message, but moved that up here instead